File: blog-sever/server-java/src/main/java/com/site/blog/config/ResponseAdvice.java
package com.site.blog.config;

import com.site.blog.constants.HttpStatusEnum;
import com.site.blog.exception.ResultException;
import com.site.blog.model.dto.Result;
import com.site.blog.util.ResultGenerator;
import lombok.extern.slf4j.Slf4j;
import org.springframework.core.MethodParameter;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.server.ServerHttpRequest;
import org.springframework.http.server.ServerHttpResponse;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.servlet.mvc.method.annotation.ResponseBodyAdvice;

import jakarta.servlet.http.HttpServletRequest;

/**
 * @author yanni
 * @date time 2022/6/17 11:44
 * @modified By:
 */
@RestControllerAdvice
@Slf4j
public class ResponseAdvice   {

    @ExceptionHandler(ResultException.class)
    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
    @ResponseBody
    public Result  exceptionHandler(HttpServletRequest req, ResultException e){
        log.error("发生业务异常！原因是：{}",e.getMessage());
        return ResultGenerator.getResultByHttp(HttpStatusEnum.INTERNAL_SERVER_ERROR,false,e.getMessage() );
    }

    /**
     * 处理空指针的异常
     * @param req
     * @param e
     * @return
     */
    @ExceptionHandler(value =NullPointerException.class)
    @ResponseBody
    public Result  exceptionHandler(HttpServletRequest req, NullPointerException e){
        log.error("发生空指针异常！原因是:",e);
        return ResultGenerator.getResultByHttp(HttpStatusEnum.INTERNAL_SERVER_ERROR,false,"空指针异常");
    }


    /**
     * 处理其他异常
     * @param req
     * @param e
     * @return
     */
    @ExceptionHandler(Exception.class)
    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
    @ResponseBody
    public Result  unknownExceptionHandler(HttpServletRequest req, Exception e){
        log.error("未知异常！原因是:",e);
        return ResultGenerator.getResultByHttp(HttpStatusEnum.INTERNAL_SERVER_ERROR,false,e.getMessage());
    }
}

File: myblog-kt/src/main/kotlin/com/site/blog/config/ResponseAdvice.kt
package com.site.blog.config

import com.site.blog.constants.HttpStatusEnum
import com.site.blog.exception.ResultException
import com.site.blog.model.dto.Result
import com.site.blog.util.ResultGenerator
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestControllerAdvice
import javax.servlet.http.HttpServletRequest

/**
 * @author yanni
 * @date time 2022/6/17 11:44
 * @modified By:
 */
@RestControllerAdvice
class ResponseAdvice {
    val log: Logger = LoggerFactory.getLogger(this.javaClass)

    @ExceptionHandler(ResultException::class)
    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
    @ResponseBody
    fun exceptionHandler(req: HttpServletRequest?, e: ResultException): Result<*> {
        log.error("发生业务异常！原因是：{}", e.message)
        return ResultGenerator.getResultByHttp(HttpStatusEnum.INTERNAL_SERVER_ERROR, false, e.message)
    }

    /**
     * 处理空指针的异常
     * @param req
     * @param e
     * @return
     */
//    @ExceptionHandler(value = [NullPointerException::class])
//    @ResponseBody
//    fun exceptionHandler(req: HttpServletRequest?, e: NullPointerException?): Result<String?> {
//        log.error("发生空指针异常！原因是:", e)
//
//        return ResultGenerator.getResultByHttp(HttpStatusEnum.INTERNAL_SERVER_ERROR, false, "空指针异常")
//
//    }

    /**
     * 处理其他异常
     * @param req
     * @param e
     * @return
     */
    @ExceptionHandler(Exception::class)
    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
    @ResponseBody
    fun unknownExceptionHandler(req: HttpServletRequest?, e: Exception): Result<*> {
        log.error("未知异常！原因是:", e)
        return ResultGenerator.getResultByHttp(HttpStatusEnum.INTERNAL_SERVER_ERROR, false, e.message)
    }
}